lib/cartopy/tests/mpl/conftest.py

```diff
@@ -15,6 +15,7 @@ def mpl_test_cleanup(request):
     with ExitStack() as stack:
         # At exit, close all open figures and switch backend back to original.
         stack.callback(plt.switch_backend, plt.get_backend())
+        stack.callback(plt.close, 'all')
```

Comment:
Does it matter whether we close all before or after the switch backend?

Reply:
Yes, if line 17 and 18 are swapped the warning is still given. Since the stack goes in reverse order, I think the flow is that on exit the plot is closed, then backend is switched to original.